js/pin.js

```diff
@@ -89,12 +89,16 @@ window.pin = (function () {
 
         if (currentPointerCoords.y < areaPointerLimits.TOP) {
           currentPointerCoords.y = areaPointerLimits.TOP;
-        } else if (currentPointerCoords.y > areaPointerLimits.BOTTOM) {
+        }
+        if (currentPointerCoords.y > areaPointerLimits.BOTTOM) {
```

Comment:
тут можно было оставить else для координат той же оси, потому что это взаимоисключающие условия для координат по одной оси
то же самое для y
```suggestion
        else if (currentPointerCoords.y > areaPointerLimits.BOTTOM) {
```